mattermost-plugin/server/plugin.go

```diff
@@ -109,7 +109,7 @@ func (p *Plugin) OnActivate() error {
 		return fmt.Errorf("error initializing the DB: %w", err)
 	}
 	if cfg.AuthMode == server.MattermostAuthMod {
-		layeredStore, err2 := mattermostauthlayer.New(cfg.DBType, sqlDB, db, logger, p.API)
+		layeredStore, err2 := mattermostauthlayer.New(cfg.DBType, sqlDB, db, logger, p.API, client)
```

Comment:
We don't need to pass it here, we can create a new client inside using the p.API, the client it is just a wrapper that provide some helpers. so passing it around all the time is not specially useful.

My suggestion here would be, or we pass the APIClient directly instead of p.API, and then use it, or we create it inside.

Probably I prefer the former, but I would do the later for this PR, and then convert it in to the other one in another PR.

Reply:
We talked it trhoguh mattermost and the fact that we also need the `p.Driver` inside, makes no sense to this change until we make a refactor to use always the APIClient all the time, and that should be in another PR.

Reply:
👍

Reply:
@jespino @harshilsharma63 , FYI we are no longer using client (plugin api wrapper) anywhere in FocalBoard.  For the multi-product architecture I've created a ServicesAPI that has all the methods needed to interact with mmserver.  There are adapters that allow the plugin API and the new product APIs to implement the ServicesAPI.